internal/middleware/logging.go

```diff
@@ -14,11 +18,24 @@ func NewLoggingInterceptor(logger logger.Logger) grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 		start := time.Now()
 
+		spanCtx := trace.SpanContextFromContext(ctx)
+		var traceID trace.TraceID
+		if !spanCtx.HasTraceID() {
+			traceID, _ = trace.TraceIDFromHex(hex.EncodeToString(ulid.Make().Bytes()))
+			spanCtx = spanCtx.WithTraceID(traceID)
+		} else {
+			traceID = spanCtx.TraceID()
+		}
+
+		ctx = trace.ContextWithSpanContext(ctx, spanCtx)
+
 		resp, err := handler(ctx, req)
 
 		fields := []zap.Field{
 			zap.String("method", info.FullMethod),
 			zap.Duration("took", time.Since(start)),
+			zap.String("trace_id", traceID.String()),
+			zap.String("raw_request", fmt.Sprintf("{%s}", req)),
```

Comment:
We should json serialize the request instead of passing it straight through. The String serialization format of the request will tend to be a pain to read.

Reply:
I tried both and it doesn't seem to make much difference. In the end both look like:
```
"raw_response": "{\"id\":\"01GQK227ACEV4KTGZJT62FV5NE\",\"name\":\"github\",\"created_at\":{\"seconds\":1674603208,\"nanos\":12914000},\"updated_at\":{\"seconds\":1674603208,\"nanos\":12914000}}"
```

In this case we can save milliseconds by not marshalling to JSON 🤷

Reply:
Fixed. Thanks!